internal/rest/dequeue_nodes.go

```diff
@@ -60,7 +60,7 @@ func (rest *RestOperations) DequeueNodes(key string) {
 		utils.AviLog.Warnf("key: %s, msg: no model found for the key", key)
 		// In the case of L7 shared VS, the following condition check makes sure the
 		// VIPs persist over AKO reboot.
-		if lib.IsShardVS(key) {
+		if lib.IsShardVS(key) && lib.GetRetainSharedVSIPs() {
```

Comment:
@swathinsankaran : If we change condition to `lib.GetRetainSharedVSIPs() && lib.IsShardVS(key)`, it will be more faster as default value to flag is `false`. So we will avoid un-necessary computation.

Reply:
@sudswasavi : Are we proposing this solution for scenario if customer wants to migrate from shardVsSize "small" to "large" or other combinations of shardVsSize?

Reply:
I have addressed the comment.